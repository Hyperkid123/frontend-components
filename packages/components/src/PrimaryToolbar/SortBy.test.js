import React from 'react';
import SortBy from './SortBy';
import { render } from '@testing-library/react';

describe('PrimaryToolbar', () => {
  describe('should render', () => {
    it('no data', () => {
      const { container } = render(<SortBy />);
      expect(container).toMatchSnapshot();
    });

    it('desc direction', () => {
      const { container } = render(<SortBy direction="desc" />);
      expect(container).toMatchSnapshot();
    });
  });

  describe('API', () => {
    it('should NOT call changeDirection', () => {
      const onSortChange = jest.fn();
      const { container } = render(<SortBy />);
      container.querySelector('button').click();
      expect(onSortChange).not.toHaveBeenCalled();
    });

    it('should call changeDirection - desc', () => {
      const onSortChange = jest.fn();
      const { container } = render(<SortBy onSortChange={onSortChange} />);
      container.querySelector('button').click();
      expect(onSortChange).toHaveBeenCalled();
      expect(onSortChange.mock.calls[0][1]).toBe('desc');
    });

    it('should call changeDirection - desc', () => {
      const onSortChange = jest.fn();
      const { container } = render(<SortBy onSortChange={onSortChange} direction="desc" />);
      container.querySelector('button').click();
      expect(onSortChange).toHaveBeenCalled();
      expect(onSortChange.mock.calls[0][1]).toBe('asc');
    });
  });
});
