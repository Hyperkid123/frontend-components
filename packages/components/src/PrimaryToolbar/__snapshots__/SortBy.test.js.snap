// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryToolbar should render desc direction 1`] = `
<Button
  onClick={[Function]}
  variant="plain"
>
  <Icon
    size="sm"
  >
    <SortAmountDownIcon />
  </Icon>
</Button>
`;

exports[`PrimaryToolbar should render no data 1`] = `
<Button
  onClick={[Function]}
  variant="plain"
>
  <Icon
    size="sm"
  >
    <SortAmountUpIcon />
  </Icon>
</Button>
`;
